src/components/Checkbox.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { StyleSheet, Text, Pressable, View } from 'react-native'
+
+import { colors } from '../config/themes/appThemes'
+
+interface CheckboxProps {
+  onChange: (value: string) => void
+  value: string
+}
+
+export const Checkbox: React.FC<CheckboxProps> = ({ onChange, value }) => {
+  const [isChecked, setIsChecked] = React.useState(false)
+
+  const toggleCheckbox = () => {
+    setIsChecked((prev) => !prev)
+    onChange(!isChecked ? 'checked' : 'unchecked')
+  }
+
+  return (
+    <Pressable style={styles.container} onPress={toggleCheckbox}>
+      <View style={[styles.checkbox, isChecked && styles.checked]} />
```

Comment:
Falto el icono de check cuando el componente es seleccionado:

```suggestion
     <View style={[styles.checkbox, isChecked && styles.checked]}>
        <StyledText neutralBase caption1 center white>
          {isChecked ? '✔' : ''}
        </StyledText>
      </View>
```